### PR TITLE
#506 display custom locktime

### PR DESCRIPTION
**Pull request by ciny**
Dates need work on embed platforms. All tests pass.

### CODE REVIEW

File: src/apps/wallet/sign_tx/layout.py [hunk 1]
```diff
@@ -65,3 +84,16 @@ def split_op_return(data):
     text = Text("Confirm sending", ui.ICON_SEND, icon_color=ui.RED)
     text.normal("Trying to spend", "coins from another chain.", "Continue?")
     return await confirm(ctx, text, ButtonRequestType.SignTx)
+
+
+async def confirm_nondefault_locktime(ctx, lock_time):
+    text = Text("Confirm locktime", ui.ICON_SEND, icon_color=ui.GREEN)
+    if lock_time < 500000000:
```
Comment (prusnak) [inline src/apps/wallet/sign_tx/layout.py]: Let's extract `500000000` as a constant similarly as `_EMBEDDED_TIMESTAMP_DIFF` is extracted.

File: src/apps/wallet/sign_tx/layout.py [hunk 2]
```diff
@@ -8,11 +11,27 @@
...
+def convert_to_embedded_timestamp(unix_timestamp):
+    return max(0, unix_timestamp - _EMBEDDED_TIMESTAMP_DIFF)
+
+
+# dates before 1.1.2000 will resolve to 1.1.2000 on non-unix
+def format_date(timestamp):
+    if not sys.platform.equals("unix"):
+        timestamp = convert_to_embedded_timestamp(timestamp)
+
+    return utime.localtime(timestamp)
```
Comment (prusnak) [inline src/apps/wallet/sign_tx/layout.py]: We might consider avoiding using the `utime` module completely and compute the date in pure Python instead.

File: src/apps/wallet/sign_tx/layout.py (hunk 1 above)
Comment (ciny) [reply src/apps/wallet/sign_tx/layout.py]: done

File: src/apps/wallet/sign_tx/layout.py (hunk 2 above)
Comment (ciny) [reply src/apps/wallet/sign_tx/layout.py]: based on what we agreed in meeting all timestamp->formatted date functionality was removed.

**Comment by prusnak** [comment]
Great, thanks!

**Pull request merged**